packages/auth/src/OAuth.ts

```diff
@@ -1,12 +1,166 @@
+import { Popup } from '@salte-auth/popup';
+import { SalteAuth } from '@salte-auth/salte-auth';
+import mitt from 'mitt';
+import { AuthParameters } from './AuthParameters';
+import CartoProvider from './CartoProvider';
+import { CARTO_AUTHORIZATION_BASE, THRESHOLD } from './constants';
+import Credentials from './Credentials';
+import { Iframe } from './Iframe';
+import UserInfo from './UserInfo';
+import { unknownScopes } from './utils';
+
 class OAuth {
-  private _clientId: string;
+  private _client: SalteAuth;
+  private _carto: CartoProvider;
+  private _refresher: Iframe;
+  private _refreshTimeout: number = -1;
```

Comment:
How about introducing a CONST?

Reply:
Sure, but this is just TS bull:hankey:, the -1 has no significance. 

I have to initialize the refreshTimeout to something, and since timeoutIDs are [positive integers](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#Return_value), -1 is a good fit.

Reply:
ok, and how about that one below?
```
    if (this._carto.expiresIn !== -1) {
      this._scheduleRefresh();
    }
```